chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -582,6 +542,87 @@ impl<'a, S: TransactionVerifierStorageRef, U: UtxosView> TransactionVerifier<'a,
         }
     }
 
+    pub fn connect_pos_accounting_outputs(
+        &mut self,
+        tx_source: TransactionSource,
+        tx: &Transaction,
+    ) -> Result<(), ConnectTransactionError> {
+        let tx_undo = tx
+            .outputs()
+            .iter()
+            .filter_map(|output| match output.purpose() {
+                OutputPurpose::StakePool(data) => Some((data, output.value())),
+                OutputPurpose::Transfer(_)
+                | OutputPurpose::LockThenTransfer(_, _)
+                | OutputPurpose::Burn => None,
+            })
+            .map(
+                |(pool_data, output_value)| -> Result<PoSAccountingUndo, ConnectTransactionError> {
+                    let tx_id = tx.get_id();
+                    let total_inputs = self.calculate_total_inputs(tx.inputs())?;
```

Comment:
This is a design flaw. The function `calculate_total_inputs` should be called exactly once, and then the returned amount should be drained through outputs. The way it stands now, there can be money printing, where in accounts, money is spent for delegation, and for utxos, the same money is spent. This has to be redesigned.